### ISSUE TITLE
Fluent-bit forward input hangs after client connection closure

### ISSUE THREAD

**Issue by andreapicaro** [fluent/fluent-bit#7796]
## Bug Report

**Describe the bug**
Fluent-bit 2.1.6 on Win10. I use two fluent-bit instances. The first instance forwards logs to the second. Whenever TCP connection is closed by client, server enters frozen state with high cpu usage and doesn't receive logs anymore. So far, system works only until the log flow is regular enough to not trigger TCP connection closure. Changing net.keepalive_idle_timeout delays the hanging by the configured amount.

**To Reproduce**
Attached file [fluent-bit-debug.zip](https://github.com/fluent/fluent-bit/files/12262440/fluent-bit-debug.zip) contains cmd script to start client and server instances and to produce test logs.
- Start first fluentbit instance configured with tail input and forward output.
- Start second fluentbit instance configured with forward input and stdout output.
- Add a few lines to input file. Their are correctly printed on server instance.
- Wait around 20 seconds, TCP connection is closed:
```
#Client
[2023/08/04 17:38:38] [debug] [upstream] drop keepalive connection #-1 to 127.0.0.1:24224 (keepalive idle timeout)
#Server
[2023/08/04 17:38:38] [trace] [input:forward:forward.0 at C:\projects\fluent-bit-2e87g\plugins\in_forward\fw_conn.c:100] fd=936 closed connection
```
- Server starts consuming high cpu.
- Any additional log sent by client is not processed by server anymore.

**Expected behavior**
Server stays idle after connection closure and keeps processing incoming logs

**Your Environment**

* Version used: fluent-bit 2.1.6 Win
* Server type and version: Reproduced on Win 10 64 bit and Windows Server 2022 Datacenter

**Comment by leonardo-albertovich** [comment]
Considering that you are running fluent-bit 2.1.6 I think this might be the bug fixed by PR #7575 which was included in fluent-bit 2.1.7 so please update to the latest version (2.1.8) and let us know if the bug persists.

**Comment by andreapicaro** [comment]
I can confirm that 2.1.8 solves the issue, thanks. 
Just a side question, i observed that first disconnection happens after 20s for all successive connections timeout is decreased to 5 seconds; is it a wanted behavior?
